controllers/userController.js

```diff
@@ -0,0 +1,37 @@
+const UserModel = require('../models/userModel');
+const UserValidationSchema = require('../validations/userValidation');
+
+const controller = {
+  showAllUsers: async (req, res) => {
+    let users = [];
+    try {
+      users = await UserModel.aggregate([{ $match: {} }, { $sort: { updatedAt: -1 } }]);
+    } catch (error) {
+      res.status(500);
+      return res.json({
+        error: 'Failed to fetch users from database',
+      });
+    }
+    res.status(200);
+    return res.json(users);
+  },
+  showUsername: async (req, res) => {
+    const username = req.params.username;
+    try {
+      const user = await UserModel.findOne({ username }).exec();
```

Comment:
Need to project the result here so that not all user data is sent to the client (i.e. no need for client to see hash, _id etc). Second parameter of findOne { _id: 0, hash: 0 } etc

Reply:
DOne the update